75 howto100m preprocessing

cwmeijer opened the pull request:
This will add howto100m preprocessing to platalea.

cwmeijer committed:
add minimal sample dataset with howto100m structure for testing

cwmeijer committed:
add video audio extraction preprocessing

cwmeijer committed:
add moviepy dependency to tox.ini

cwmeijer committed:
extract howto100m audio features

cwmeijer committed:
add sndfile to tox dependecies

cwmeijer committed:
Update pythonapp.yml

cwmeijer committed:
move test files

cwmeijer committed:
add cli argument

bhigy commented on platalea/utils/preprocessing.py:
```diff
@@ -17,13 +21,38 @@
+
+def extract_howto100m_audio_features(dataset_path, audio_subdir, feat_config):
+    audio_dir_path = dataset_path / audio_subdir
+    file_names = os.listdir(audio_dir_path)
```
as `audio_dir_path` is a `pathlib.Path`, you can directly do `audio_dir_path.iterdir()`

cwmeijer replied:
Thanks, I'm not used to using pathlib.Path. This is a nice helper function.

bhigy commented on .github/workflows/pythonapp.yml:
```diff
@@ -15,6 +15,8 @@ jobs:
         python-version: 3.8
     - name: Install tox and coverage
       run: pip install tox coverage
+    - name: Install libsndfile1-dev
```
is that related to moviepy?

cwmeijer replied:
yeah, it's in the documentation of moviepy. It won't install otherwise.

cwmeijer committed:
add howto100m cli arguments

cwmeijer committed:
add defaults for howto100m cli arguments

bhigy commented on platalea/utils/preprocessing.py:
```diff
@@ -17,13 +21,38 @@
+    file_names = os.listdir(audio_dir_path)
+    paths = [audio_dir_path / fn for fn in file_names]
+    features = audio_features(paths, feat_config)
+    torch.save(dict(features=features, filenames=file_names), dataset_path / 'mfcc_features.pt')
```
aren't we supposed to save that as a numpy memmap?

cwmeijer replied:
good point of course

cwmeijer committed:
fix path object

cwmeijer committed:
specify change location in test error message

cwmeijer committed:
use iterdir helper function

cwmeijer committed:
save audio features in compressed np format

egpbos commented:
Is this ready to merge?

egpbos commented on requirements.txt:
```diff
@@ -1,3 +1,4 @@
+moviepy
```
This should probably be in setup.py, so here it can be removed. `sndfile` should also be in setup.py (not sure if it is).

We decided at some point that we would use requirements.txt only to pin specific dependency versions for reproducibility of specific experiments as published in some paper or at a conference.

cwmeijer replied:
I've moved those dependencies around between setup and even tox because of your suggestion below but it kept giving me errors. I'm sure something can be streamlined here, but I could not make it work the way you suggested (which would have been the most clean way, I agree)

egpbos commented on tox.ini:
```diff
@@ -12,6 +12,8 @@ deps =
     scikit-learn
     python-Levenshtein
+    moviepy
+    sndfile 
```
These two (moviepy and sndfile) can be removed here. If they are in setup.py, they will automatically be installed via pip in the tox run.

egpbos replied:
Actually, I think scikit-learn and python-Levenshtein should also be removed here, they should also be installed through pip via setup.py. Could you try that?

egpbos reviewed (changes requested):
Some dependency reshuffling, otherwise LGTM.

cwmeijer committed:
change howto100m audio feature format to memmap

cwmeijer commented:
Thanks for reviewing. I'll take it into account. I put this thing back into draft mode though. I'm at the same time working on the data processing (as opposed to this PR about PREprocessing). Many problems that I encounter in the processing, can best be solved by changing the preprocessing. When I think it all makes sense and is functional, I'll ask you (or 1 of you) to have another look at the final changes.

cwmeijer committed:
add id_map containing memmap indices to preprocessing

cwmeijer committed:
fix linter suggested style changes

cwmeijer committed:
Merge branch 'master' into 75-howto100m-preprocessing

cwmeijer committed:
fix suggestions by linter

cwmeijer committed:
Merge branch '75-howto100m-preprocessing' of https://github.com/spokenlanguage/platalea into 75-howto100m-preprocessing

cwmeijer committed:
save s3d_features file with audio meta data

cwmeijer committed:
add s3d features to test data

cwmeijer committed:
change id_map to index list

cwmeijer committed:
pin torch versions (see #116)

cwmeijer committed:
pin torch versions (see #116)

cwmeijer committed:
Merge branch 'master' into 75-howto100m-preprocessing

cwmeijer commented:
I fixed the issues raised by Patrick some time ago. I also changed many other things. For instance, there's an index file being written that contains some metadata of start/end locations in the memmap and paths to video features. I think it's ready for merge now, so I requested another review.
